apps/backend/src/app/api/latest/integrations/credential-scanning/revoke/route.tsx

```diff
@@ -0,0 +1,186 @@
+
+
+    if (updatedApiKey.projectUserId) {
+      // For user API keys, notify the user
+
+      const projectUser = await prismaClient.projectUser.findUnique({
+        where: {
+          tenancyId_projectUserId: {
+            tenancyId: updatedApiKey.tenancyId,
+            projectUserId: updatedApiKey.projectUserId,
+          },
+        },
+        include: {
+          contactChannels: true,
+        },
+      });
+
+      if (!projectUser) {
+        // This should never happen
+        throw new StackAssertionError("Project user not found");
+      }
+      // We might have other types besides email, so we disable this rule
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+      const primaryEmail = projectUser.contactChannels.find(c => c.type === 'EMAIL' && c.isPrimary)?.value ?? undefined;
+      if (primaryEmail) {
+        affectedEmails.add(primaryEmail);
+      }
+    } else if (updatedApiKey.teamId) {
+      // For team API keys, notify users with manage_api_keys permission
+
+      const userIdsWithManageApiKeysPermission = await prismaClient.$transaction(async (tx) => {
+        const tenancy = await getTenancy(updatedApiKey.tenancyId);
+
+        if (!tenancy) {
+          throw new StackAssertionError("Tenancy not found");
+        }
+
+        if (!updatedApiKey.teamId) {
+          throw new StackAssertionError("Team ID not specified in team API key");
+        }
+
+        const permissions = await listUserTeamPermissions(tx, {
+          tenancy,
+          teamId: updatedApiKey.teamId,
+          permissionId: '$manage_api_keys',
+          recursive: true,
+        });
+
+        return permissions.map(p => p.user_id);
+      });
+
+
+      const usersWithManageApiKeysPermission = await prismaClient.projectUser.findMany({
+        where: {
+          tenancyId: updatedApiKey.tenancyId,
+          projectUserId: {
+            in: userIdsWithManageApiKeysPermission,
+          },
+        },
+        include: {
+          contactChannels: true,
+        },
+      });
+
+      for (const user of usersWithManageApiKeysPermission) {
+        // We might have other types besides email, so we disable this rule
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        const primaryEmail = user.contactChannels.find(c => c.type === 'EMAIL' && c.isPrimary)?.value ?? undefined;
+        if (primaryEmail) {
+            affectedEmails.add(primaryEmail);
+        }
+      }
+    }
+
+
+    // Create email content
+    const subject = `API Key Revoked: ${updatedApiKey.description}`;
+    const htmlContent = `
+      <div style="font-family: Arial, sans-serif; max-width: 600px; margin: 0 auto; padding: 20px;">
+        <h2 style="color: #333;">API Key Revoked</h2>
+        <p style="color: #555; font-size: 16px; line-height: 1.5;">
+          Your API key "${updatedApiKey.description}" has been automatically revoked because it was found in a public repository.
```

Comment:
```suggestion
          Your API key "${updatedApiKey.description}" for ${project.name} has been automatically revoked because it was found in a public repository.
```